addons/mail/static/src/components/chatter/chatter_tests.js

```diff
@@ -452,6 +461,116 @@ QUnit.test('do not post message with "Enter" keyboard shortcut', async function
+                            ],
+                        ]
+                    });
+            }
+            return this._super(...arguments);
+        },
+    });
+    const chatter = this.env.models['mail.chatter'].create({
+        threadId: 100,
+        threadModel: 'res.partner',
+    });
+    await this.createChatterComponent({ chatter });
+
+    await afterNextRender(() =>
+        document.querySelector(`.o_ChatterTopbar_buttonSendMessage`).click()
+    );
+
+    assert.containsOnce(
+        document.body,
+        '.o_ComposerSuggestedRecipientList',
+        "Must containt a suggested recipient list"
+    );
+    assert.containsN(
+        document.body,
+        '.o_ComposerSuggestedRecipient',
+        3,
+        "at start, the list should not contain more than 3 suggested recipients"
+    );
+    const title = document.querySelector('.o_ComposerSuggestedRecipientList .o_ComposerSuggestedRecipient:first-child').getAttribute('title');
+    assert.strictEqual(
+        title,
+        "Add as recipient and follower (reason: CC email)",
+        "reason must be present in the suggested recipient title",
+    );
+    const checkboxUnchecked = document.querySelector('.o_ComposerSuggestedRecipientList .o_ComposerSuggestedRecipient:first-child input[type=checkbox]');
+    assert.notOk(
+        checkboxUnchecked.checked,
+        "suggested recipient without a partner should not be checked by default",
+    );
+    const checkboxChecked = document.querySelector('.o_ComposerSuggestedRecipientList .o_ComposerSuggestedRecipient input.o-has-partner');
+    assert.ok(
+        checkboxChecked.checked,
+        "suggested recipient with a partner should be checked by default.",
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ComposerSuggestedRecipientList_showMore',
+        "suggested recipient list should containt a \"show more\" button."
+    );
+
+    await afterNextRender(() =>
+        document.querySelector(`.o_ComposerSuggestedRecipientList_showMore`).click()
+    );
+    assert.containsN(
+        document.body,
+        '.o_ComposerSuggestedRecipient',
+        4,
+        "suggested recipient list should display all the suggested recipients."
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ComposerSuggestedRecipientList_showLess',
+        "suggested recipient list should containt a \"show less\" button."
+    );
+
+    await afterNextRender(() =>
+        document.querySelector(`.o_ComposerSuggestedRecipientList_showLess`).click()
+    );
+    assert.containsN(
+        document.body,
+        '.o_ComposerSuggestedRecipient',
+        3,
+        "suggested recipient list should display 3 suggested recipients after clicking on \"show less\"."
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ComposerSuggestedRecipientList_showMore',
+        "suggested recipient list should containt a \"show More\" button after clicking on \"show less\"."
+    );
```

Comment:
This behaviour should be in (an)other test(s).
Current test is too long and tests too many things. Any change/improvements on suggested recipients, we are encouraged to extend this test and eventually we will have a single test with hundreds of assertions... I'm exaggerating of course...